spec/micro/README.md

```diff
@@ -0,0 +1,770 @@
+- Establishes the ordering of the channel
+- Establishes unique identifiers for the applications on either chain to use to reference each other when sending and receiving packets.
+- The application protocol can be continually upgraded over time by using the upgrade handshake which allows the same channel which may have accumulated state to use new mutually agreed upon application packet data format(s) and associated new logic.
+- Ensures exactly-once delivery of packet flow datagrams (Send, Receive, Acknowledge, Timeout)
+- Ensures valid packet flow (Send => Receive => Acknowledge) XOR (Send => Timeout)
+
+### Identifying Counterparties
+
+In core IBC, the connection and channel handshakes serve to validate the clients are valid clients of the counterparty, ensure the IBC version and application versions are mutually compatible, as well as providing unique identifiers for each side to refer to the counterparty.
+
+Since we are removing handshakes in IBC lite, we must have a different way to provide the chain with knowledge of the counterparty. With a client, we can prove any key/value path on the counterparty. However, without knowing which identifier the counterparty uses when it sends messages to us; we cannot differentiate between messages sent from the counterparty to our chain vs messages sent from the counterparty with other chains. Most implementations will not be able to store the ICS-24 paths directly as a key in the global namespace; but will instead write to a reserved, prefixed keyspace so as not to conflict with other application state writes. Thus the counteparty information we must have includes both its identifier for our chain as well as the key prefix under which it will write the provable ICS-24 paths.
+
+Thus, IBC lite will introduce a new message `ProvideCounterparty` that will associate the counterparty client of our chain with our client of the counterparty. Thus, if the `ProvideCounterparty` message is submitted to both sides correctly. Then both sides have mirrored <client,client> pairs that can be treated as channel identifiers. Assuming they are correct, the client on each side is unique and provides an authenticated stream of packet data between the two chains. If the `ProvideCounterparty` message submits the wrong clientID, this can lead to invalid behaviour; but this is equivalent to a relayer submitting an invalid client in place of a correct client for the desired chain. In the simplest case, we can rely on out-of-band social consensus to only send on valid <client, client> pairs that represent a connection between the desired chains of the user; just as we currently rely on out-of-band social consensus that a given clientID and channel built on top of it is the valid, canonical identifier of our desired chain.
+
+```typescript
+interface Counterparty {
+    channelId: Identifier
```

Comment:
I am not entirely sure in this spec we should keep using channelIDs instead of clientIDs. Is this an implementation related detail to ensure compatibility with ibc-go? If yes, shouldn't we use clientIDs in the spec and, additionally, provide a comment stating that the implementation we will be using channelIDs?